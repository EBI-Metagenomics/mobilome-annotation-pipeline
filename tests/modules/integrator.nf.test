nextflow_process {

    name "Test Process INTEGRATOR"
    script "modules/integrator.nf"
    process "INTEGRATOR"

    test("The intagrator moduel should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/tests/test_data/integrator_inputs/contigs.gff")
                input[1] = file("${projectDir}/tests/test_data/integrator_inputs/contigID.map")
                input[2] = file("${projectDir}/tests/test_data/integrator_inputs/1kb_contigs.fasta.tsv")
                input[3] = file("${projectDir}/tests/test_data/integrator_inputs/palidis_info.txt")
                input[4] = file("${projectDir}/tests/test_data/integrator_inputs/5kb_contigs.summary")
                input[5] = [
                    file("${projectDir}/tests/test_data/integrator_inputs/contig_1.gbk"),
                    file("${projectDir}/tests/test_data/integrator_inputs/contig_10.gbk"),
                    file("${projectDir}/tests/test_data/integrator_inputs/contig_6.gbk")
                ]
                input[6] = file("${projectDir}/tests/test_data/integrator_inputs/icf_concat.summary")
                input[7] = file("${projectDir}/tests/test_data/integrator_inputs/icf_dr.txt")
                input[8] = file("${projectDir}/tests/test_data/integrator_inputs/mobileOG.tsv")
                input[9] = file("${projectDir}/tests/test_data/integrator_inputs/5kb_contigs_virus_summary.tsv")
                input[10] = file("${projectDir}/tests/test_data/integrator_inputs/5kb_contigs_plasmid_summary.tsv")
                input[11] = file("${projectDir}/tests/test_data/integrator_inputs/virify_hq.gff")
                input[12] = file("${projectDir}/tests/test_data/integrator_inputs/Crisprs_REPORT.tsv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
