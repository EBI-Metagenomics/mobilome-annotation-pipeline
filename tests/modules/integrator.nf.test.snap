{
    "The intagrator moduel should run without failures": {
        "content": [
            {
                "0": [
                    "mobilome_prokka.gff:md5,9eed79e04fb0fbca9fb82b9b9093a03c"
                ],
                "1": [
                    "overlapping_integrons.txt:md5,ec0cb32fbf0260ddf32be7074f5c71a2"
                ],
                "2": [
                    "discarded_mge.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "mobilome_prokka_gff": [
                    "mobilome_prokka.gff:md5,9eed79e04fb0fbca9fb82b9b9093a03c"
                ]
            }
        ],
        "timestamp": "2023-11-28T21:26:52.54156"
    }
}